amgut/static/js/interactive.js

```diff
@@ -0,0 +1,350 @@
+    var other = {label: 'Other', data: collapsed[sorted[loop][0]].data,
+                 backgroundColor: 'rgba(180,180,180)' };
+    for (var i = loop + 1; i < sorted.length; i++) {
+      other.data = sumArrays(other.data, collapsed[sorted[i][0]].data);
+    }
+    for (var i = 0; i < other.data.length; i++) {
+      other.data[i] = Math.floor(other.data[i] * 10000) / 10000;
+    }
+    further_collapsed.push(other);
+  }
+  if (filter_to !== null) {
+    further_collapsed = filterSites(further_collapsed, filter_to);
+  }
+  return further_collapsed;
+}
+
+function calcFoldChange(newData, rawData, level, dataPos) {
+  var foldChanges = [];
+  var collapsed = collapse(rawData, level, 10000);
+  var dataCollapsed = collapse(newData, level, 10000);
+  for (var i = 0; i < dataCollapsed.length; i++) {
+    var otu = dataCollapsed[i];
+    var pos = findOtu(collapsed, otu.label);
+    //sanity checks to make sure we aren't doing something dumb
+    if (pos === -1 || otu.data === 0) {
+      continue;
+    }
+    var sample_data = collapsed[pos].data;
+
+    //Compute the fold changes, making them negative when needed
+    var val = sample_data[dataPos] / otu.data;
+    if (val === 0) {
+      continue;
+    }
+    foldChanges.push([otu.label, Math.floor(Math.log2(val) * 100) / 100]);
+  }
+  //Sort by the average and build the data information
+  var sorted = foldChanges.sort(function(a, b) {
+    return b[1] - a[1];
+  });
+  foldData = [];
+  labels = [];
+  for (var i = 0; i < sorted.length; i++) {
+    //TODO: Some way to filter significance
+    foldData.push(sorted[i][1]);
+    labels.push(sorted[i][0]);
+  }
+  return [labels, foldData];
+}
+
+function fold_change() {
+  var level = $('#collapse-fold').val();
+  var category = $('#meta-cat-fold').val();
+  var barcode = $('#barcode-fold').val();
+  //Set alpha diversity image going first
+  if (category === '') {
+    $('#alpha-div-image').attr('src', '/interactive/alpha_div/' + barcode);
+  } else {
+      $('#alpha-div-image').attr('src', '/interactive/alpha_div/' + barcode +
+        '?category=' + category);
+  }
+
+  $.get('/interactive/metadata/', {category: category, barcode: barcode})
+    .done(function(data) {
+      var dataPos = barChartFoldData.barcodes.findIndex(function(y) {
+        return barcode == y;
+      });
+      var rawData = barChartSummaryData.rawData;
+      foldChanges = calcFoldChange(data, rawData, level, dataPos);
+      var labels = foldChanges[0];
+      var foldData = foldChanges[1];
+      barChartFoldData.labels = labels;
+      barChartFoldData.datasets = [{
+        backgroundColor: 'rgba(151,187,205,0.7)',
+        strokeColor: 'rgba(151,187,205,0.8)',
+        highlightBackground: 'rgba(220,220,220,1)',
+        highlightStroke: 'rgba(220,220,220,1)',
+        label: 'fold changes',
+        data: foldData
+      }];
+      window.foldBar.update();
+    })
+    .fail(function() {
+      alert('FAIL!');
```

Comment:
this is all still placeholder text. There will be another PR following this one to get text in. This is just for functionality right now.